mod-data-export/src/test/resources/domain/dataexport/features/export.feature

```diff
@@ -0,0 +1,144 @@
+    And request fileDefinition
+    When method POST
+    Then status 201
+    And match response.status == 'NEW'
+    And match response.uploadFormat == '<uploadFormat>'
+
+    #should return created file definition
+    Given path 'data-export/file-definitions', <fileDefinitionId>
+    When method GET
+    Then status 200
+    And match response.status == 'NEW'
+    And match response.uploadFormat == '<uploadFormat>'
+
+    #should upload file by created file definition id
+    Given path 'data-export/file-definitions/',<fileDefinitionId>,'/upload'
+    And configure headers = headersUserOctetStream
+    And request karate.readAsString('classpath:samples/file-definition/<fileName>')
+    When method POST
+    Then status 200
+    And match response.jobExecutionId == '#present'
+    And match response.status == 'COMPLETED'
+    And match response.uploadFormat == '<uploadFormat>'
+    And match response.sourcePath == '#present'
+    And def jobExecutionId = response.jobExecutionId
+    And call pause 1000
+
+    #should export instances and return 204
+    Given path 'data-export/export'
+    And configure headers = headersUser
+    And def requestBody = {'fileDefinitionId':'#(<fileDefinitionId>)','jobProfileId':'#(defaultJobProfileId)'}
+    And request requestBody
+    When method POST
+    Then status 204
+    And call pause 7000
+
+    #should return job execution by id
+    Given path 'data-export/job-executions'
+    And param query = 'id==' + jobExecutionId
+    When method GET
+    Then status 200
+    And match response.jobExecutions[0].status == 'COMPLETED'
+    And match response.jobExecutions[0].progress == {exported:1, failed:0, total:1}
+    And def fileId = response.jobExecutions[0].exportedFiles[0].fileId
+
+    #should return download link for instance of uploaded file
+    Given path 'data-export/job-executions/',jobExecutionId,'/download/',fileId
+    When method GET
+    Then status 200
+    And match response.fileId == '#notnull'
+    And match response.link == '#notnull'
+    * def downloadLink = response.link
+
+    #download link content should not be empty
+    Given url downloadLink
+    When method GET
+    Then status 200
+    And match response == '#notnull'
+
+    Examples:
+      | fileName        | uploadFormat | fileDefinitionId    |
+      | test-export-csv.csv | csv          | csvFileDefinitionId |
+      | test-export-cql.cql | cql          | cqlFileDefinitionId |
+
+  Scenario: error logs should be empty after successful scenarios
+    Given path 'data-export/logs'
+    When method GET
+    Then status 200
+    And match response.totalRecords == 0
+
+  Scenario: Should return transformation fields
+    Given path 'data-export/transformation-fields'
+    When method GET
+    Then status 200
+    And assert response.transformationFields.length > 0
+
+  Scenario: should clean expired jobs if exist and return 204
+    Given path 'data-export/expire-jobs'
+    When method GET
```

Comment:
Expired jobs here refers to the ones created by `Scenario Outline: test upload file and export flow.`, right?

If yes, we should make sure that the order is preserved, but my personal preference would be to extend this test so that it first creates some jobs and than verifies that they removed.

It is up to you to decide.

Reply:
Actually, this scenario doesn't relate to the export flow and the test defined separately from the main flow in "scenario outline". I removed the scenario cause it requires a lot of time to make the job expired. This functionality covered by rest assured tests inside the module and the logic of cleaning jobs/files doesn't contain dependencies from external calls, therefore it can be dropped from integration tests I think. Your thoughts?